### ISSUE TITLE
Looping pipeline?

### ISSUE THREAD

**Issue by mjulstein** [WolfgangOfner/AzureDevOpsGitCommit#1]
Nice demo. I am working on something similar and was wondering if this will trigger the pipeline again?
Is the pipeline interrupted if there were any steps after the push to master?

**Comment by WolfgangOfner** [comment]
If you push commit to the Master branch, the pipeline will run. If you push another commit while the pipeline is still running, then another pipeline will be scheduled. This could lead to problems though because both runs want to checkout and commit code which could result in an error.

**Comment by mjulstein** [comment]
So if the pipeline pushes the pipeline commit to master, it will trigger the pipeline to run again?
Should there maybe be an initial job to check if the last commit was not made by the pipeline so the pipeline doesn't trigger itself?

**Comment by WolfgangOfner** [comment]
No the pipeline doesn't trigger another pipeline run. This would cause an infinite loop. Only developer commits cause a pipeline to run.

**Comment by mjulstein** [comment]
I'm curious to how that works, I had to put a step in the beginning of my pipeline to check if the last committer was the pipeline, and then skip the rest of the pipeline. It would be ideal if that was not necessary.
I set my pipeline up as a multi stage, where I do validating in the first stage and build/test etc + commit&push in second.
One of my steps in the first stage is to check if the last commit was made by the pipeline, but if there is a way to remove that step and not have to trigger CI at all that would be great to know. Is there a setting in Azure, build/branch policy? Maybe it could be added to the readme.md.
